core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -1418,6 +1418,29 @@ private void handleOffsetFetch(@Nullable SqlNode offset, @Nullable SqlNode fetch
       if (node instanceof SqlMerge) {
         validatingSqlMerge = true;
       }
+      if (node instanceof SqlJoin) {
+        // select * from "scott".dept,
+        //     lateral (select count(cast(deptno as integer)) from "scott".emp
+        //         where emp.deptno = dept.deptno);
+        // ==》
+        // select * from "scott".dept left join
+        //     lateral (select count(cast(deptno as integer)) from "scott".emp
+        //        where emp.deptno = dept.deptno) on true;
+        SqlJoin sqlJoin = (SqlJoin) node;
+        if (sqlJoin.getJoinType() == JoinType.COMMA && sqlJoin.getRight() instanceof SqlCall) {
```

Comment:
can you explain where the discussion in the JIRA is reflected in this code?
I think the consensus there was that a left join is not always right

Reply:
I agree, @mihaibudiu. The jira cases discusses how converting an inner lateral join to a left join is WRONG. Do not merge this PR.

Reply:
I will recheck this. This repair method does have some issues. I will add more test cases to verify it.